salvo/src/lib/api/docker_volume.proto

```diff
@@ -0,0 +1,21 @@
+syntax = "proto3";
+
+package salvo;
+
+// This message defines the properties for a given mount. It is used
+// to generate the dictionary specifying volumes for the Python Docker
+// SDK
+message VolumeProperties {
+  // Defines a list of properties governing the mount in the container
+  string bind = 1;
+
+  // Define whether the mount point is read-write or read-only
+  string mode = 2;
+}
+
+// This message defines the volume structure consumed by the command
+// to run a docker image.
+message Volume {
+  // Specify a map of volumes and their mount points for use in a container
+  map<string, VolumeProperties> volumes = 1;
```

Comment:
Do we have a good reason to use a map? Or could we just go with a repeated field of a message that includes both the string (I am guessing it is a name) and the volume properties?

Proto maps are a cool feature that helps to optimize code. I.e. we would use it if there are hundreds of volumes and we need to quickly look one up by name. If we are dealing with a much smaller set, we should not try to optimize prematurely since it comes with a cost. The serialization order of proto maps is undefined so this will make our life harder in any sort of unit tests and integration tests we will end up writing.